File: docs-examples/csfle-kotlin/src/test/kotlin/io/micronaut/mongodb/docs/NamedClientService.kt
package io.micronaut.mongodb.docs

// tag::imports[]
import com.mongodb.reactivestreams.client.MongoClient
import jakarta.inject.Inject
import jakarta.inject.Named
import jakarta.inject.Singleton
import org.reactivestreams.Publisher
// end::imports[]

// tag::clazz[]
@Singleton
class NamedClientService {

    @Inject
    @field:Named("another")
    lateinit var mongoClient: MongoClient

    fun databaseNames(): Publisher<String> = mongoClient.listDatabaseNames()
}
// end::clazz[]

File: docs-examples/csfle-python/src/test/python/micronaut/mongodb/docs/NamedClientService.py
# tag::imports[]
from typing import Annotated

from com.mongodb.reactivestreams.client import MongoClient
from jakarta.inject import Inject, Named, Singleton
from org.reactivestreams import Publisher
# end::imports[]


# tag::clazz[]
@Singleton
class NamedClientService:
    mongo_client: Annotated[MongoClient, Inject, Named("another")]

    def database_names(self) -> Publisher[str]:
        return self.mongo_client.listDatabaseNames()
# end::clazz[]
